runtime/src/accounts.rs

```diff
@@ -509,13 +509,21 @@ impl Accounts {
     }
 
     /// Once accounts are unlocked, new transactions that modify that state can enter the pipeline
-    pub fn unlock_accounts(&self, txs: &[Transaction], results: &[Result<()>]) {
+    pub fn unlock_accounts(
+        &self,
+        txs: &[Transaction],
+        txs_iteration_order: Option<&[usize]>,
+        results: &[Result<()>],
+    ) {
         let mut account_locks = self.account_locks.lock().unwrap();
         let credit_only_locks = self.credit_only_account_locks.clone();
         debug!("bank unlock accounts");
-        txs.iter().zip(results.iter()).for_each(|(tx, result)| {
```

Comment:
Bug was here.  The `results` list is ordered based on the randomized transaction order, but `txs` was not, so we were potentially unlocking the wrong accounts